docs/src/content/docs/book/functions.mdx

```diff
@@ -114,6 +116,167 @@ native storeUint(s: Builder, value: Int, bits: Int): Builder;
+asm fun wrongMyCode() { "MYCODE" }
+
+// Invokes the TVM instruction MYCODE during the compute phase,
+// which returns the contract code as a Cell
+asm fun myCode(): Cell { MYCODE }
+```
+
+The syntax for parameters and return values is the same as for other function kinds, but there is one caveat — argument values are pushed to the stack before the function body is executed, and return values are what's left on the stack afterward.
+
+Since the bodies of `asm{:tact}` functions do not contain Tact statements, any direct references to parameters in function bodies will be recognized as [TVM][tvm] instructions, which can easily lead to very obscure error messages.
+
+```tact
+/// Simply returns back the value of `x`
+asm fun identity(x: Int): Int { }
+
+/// COMPILATION ERROR!
+/// The `BOC` is not recognized as a parameter,
+/// but instead is interpreted as a non-existent TVM instruction
+asm fun bocchiThe(BOC: Cell): Cell { BOC }
+
+/// Loads a signed `len`-bit integer from Slice `s`,
+/// and returns it with the remainder of `s`
+asm fun sliceLoadInt(s: Slice, len: Int): IntSlice { LDIX }
+//                   ↑         ↑
+//                   |         Pushed last, sits on top of the stack
+//                   Pushed first, sits on the bottom of the stack
+
+/// Maps onto values placed by LDIX on the stack
+struct IntSlice { a: Int; b: Slice }
+//                ↑       ↑
+//                |       Pushed last, sits on top of the stack
+//                Pushed first, sits on the bottom of the stack
+```
+
+The return values are provided bottom-up from the stack and the unused values are discarded.
+
+```tact
+// Same function as before, but now we don't use the `IntSlice` Struct
+// and instead only take one value from the stack (going bottom-up)
+asm fun sliceLoadInt(s: Slice, len: Int): Int { LDIX }
+//                                        ↑
+//                                        captures the Int value, discarding
+//                                        the Slice one produced by LDIX instruction
+```
+
+### Arrangements {#asm-arrangements}
+
+Sometimes it's useful to change the order of arguments pushed to the stack or the order of return values. You can do that with `asm{:tact}` arrangements in the following manner:
+
+```tact
+// Changing the order of arguments to match the STDICT signature:
+// `c` will be pushed first and get on the bottom of the stack,
+// while `self` will be pushed last and get on top of the stack
+asm(c self) extends fun asmStoreDict(self: Builder, c: Cell?): Builder { STDICT }
+
+// Changing the order of return values of LDVARUINT16,
+// capturing only the last one as the return value of the whole function
+asm(-> 1 0) extends mutates fun asmLoadCoins(self: Slice): Int { LDVARUINT16 }
+//     ---
+//     Notice, that return values are best thought as tuples with indexed access into them
+//     and not as bottom-up representation of stack values
```

Comment:
Probably we need to clarify this in a more detailed example (tutorial-like), because the connection with the stack is very vague and confusing. I can do such detailed example in a separate page, and then I can add a link in here to such page. I can work on that example as a separate PR, since now I am working on writing docs. What do you think? Do you suggest I should use the same TVM instructions you were using in these examples or should I use another TVM instruction that produces at least 3 or more results to exemplify the use of structs?